File: Frontend/src/app/core/models/Purchase.ts
export class Purchase {
  id: number = -1;
  userID: number = -1;
  orderDate: Date = new Date();
  total: number = -1;
}

export class PurchaseInsert {
  userID: number = -1;
  orderDate: Date = new Date();
  total: number = -1;
}

export class PurchaseUpdate {
  id: number = -1;
  userID: number = -1;
  orderDate: Date = new Date();
  total: number = -1;
}

File: API/Models/Purchase.cs
namespace API.Models
{
    public class Purchase
    {
        public long id { get; set; } = -1;
        public long userID { get; set; } = -1;
        public DateTime orderDate { get; set; }
        public decimal total { get; set; } = -1;
    }
    public class PurchaseInsert
    {
        public long userID { get; set; } = -1;
        public DateTime orderDate { get; set; }
        public decimal total { get; set; } = -1;
    }
    public class PurchaseUpdate
    {
        public long id { get; set; } = -1;
        public long userID { get; set; } = -1;
        public DateTime orderDate { get; set; }
        public decimal total { get; set; } = -1;
    }
}